stackslib/src/net/relay.rs

```diff
@@ -6003,22 +5996,8 @@ pub mod test {
 
             let sortdb = peer.sortdb.take().unwrap();
             let mut node = peer.stacks_node.take().unwrap();
-            match Relayer::process_new_anchored_block(
```

Comment:
Why is this dropped?

Reply:
won't be dropped anymore since we removed the epoch gating logic

Reply:
Actually I'm wrong on this one
`make_anchored_block_and_microblock_from_txs` this function filters the incorrect transaction (skips it) because it has new gating logic for transaction processing, therefore the final block won't be rejected
That's why I removed the logic for block processing and checked the amount of transactions left instead, but let me know if you have better ideas on how we can modify this part

Reply:
In that case, this must be restored.  The code here is checking that block-processing catches blocks with invalid Stacks transactions.  This snippet of test code performs that check.

Reply:
@jcnelson the incorrect transaction is filtered in the step before, `make_anchored_block_and_microblock_from_txs ` this function is being called in the `make_anchored_block_from_txs` function
Which has inside `match builder.try_mine_tx` which rejects the transaction and does not include it in the end because of the gating logic (Error::InvalidStacksTransaction)
Therefore the code I removed won't have an error inside, while I added a line that checks if the incorrect tx was filtered
Do you think we should somehow modify the `make_anchored_block_and_microblock_from_txs ` function or make a new one for this particular autotest? Would be great to get more info on that, thanks

Reply:
As I said before, the purpose of this dropped code is to verify that block-processing catches blocks with invalid Stacks transactions.  If your changes make it so that it's no longer possible to construct Stacks blocks with invalid transactions such that this test continues to pass as-is, then the correct thing to do is to _fix your patch_ instead of delete my tests. 
 The function `try_mine_tx()` should not be performing an epoch gate check.

The way the system has historically prevented transactions from an as-of-yet-unsupported epoch from finding their way into blocks is to prevent them from entering the mempool.  I see you have already added the gating logic to the function `can_include_tx()`, which is sufficient.